### DISCUSSION TITLE
Revamping the existing notification system in Tasking Manager

### DISCUSSION THREAD

**Discussion by Ichchhie** [hotosm/tasking-manager#5567] [Ideas]
Several months ago, at one of our past tech collective meetups, we discussed an issue related to providing project managers with regular email updates on their projects (issue #[1377](https://github.com/hotosm/tasking-manager/issues/1377)). During that conversation, two different thoughts were presented:
1. Will adding these emails overload the project manager's email and decrease its importance?
2. What about dividing the notifications into emails and a notification section within TM?

Quite a discussion already happened in (issue #[1377](https://github.com/hotosm/tasking-manager/issues/1377)) itself around point no 1 but for no 2, what we realized is despite the presence of a notification menu 🔔 in TM, it's being unused. There is an open issue #5500 for this. We also did an analysis of notification system usage, which showed that:

1. More than 95% of the messages are left unread
2. A total of 11 types of notifications are sent from Tasking Manager
3. Over 70% of notifications are sent to users to notify them that their task has been validated.
4. But the percentage of unread count is very high compared to read ones

The complete analysis results can be viewed [here](https://www.canva.com/design/DAFZxDZnKB8/bmQDMMZLDrBGO84uxfm4HQ/view?utm_content=DAFZxDZnKB8&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton), however these results have raised questions like:

1.  Why are so many notifications left unread?
6. Identify the relevance and importance of every notification type.
7. How can the notification overload be addressed? How relevant are task-validated notifications covering more than 70% of total notifications?
8. What can increase user engagement with the notification system?
10. Do users have full control over the notifications they want to get? Considering that the majority of notifications are sent by email and application. Should the level of control be increased?

So, opening this discussion to talk about the existing notification system, its pain point, and probable ideas to make it more prominent and usable. Excited to hear all of your thoughts!

**Comment by ramyaragupathy** [comment]
Notes from the collective meetup today:
*  users leave messages unread on TM even if they have read all of them within the application
*  some users simply click all the notifications and mark them as read 
*  some users have the messages read via email, so they leave TM notifications unread
* users see value in validation notification only if there is a comment/feedback added to the action. Otherwise, plain validated message is delivering no value
*  prioritisng messages for different groups of users - For project creators/managers having notifications prioritized that are in the 'questions and comments' section of a project is very useful & important for the project's quality and feedback
 

cc @Ichchhie @petya-kangalova @Aadesh-Baral @HelNershingThapa @emi420 @d-rita

**Reply by ramyaragupathy** [reply]
* make it default for all mappers
* avoid duplication of notifications
* some users also read notification from the popup (and then delete the notification without "reading")

**Reply by ramyaragupathy** [reply]
Fleshing out one of the items for implementation - https://github.com/hotosm/tasking-manager/issues/5944

**Comment by Ichchhie** [comment]
I realized the notification message for mapper level upgrade is not that helpful both content and design-wise!


![Screenshot_896](https://user-images.githubusercontent.com/33247422/219340358-a854e6a5-90a4-47a1-8dab-c8bd0207174b.png)

I would prefer it to say like "Congratulations, you have been upgraded to Intermediate level. Kudos, keep mapping" kind of encouraging message.
![Congratulations, you have been upgraded to Intermediate level  Keep mapping](https://user-images.githubusercontent.com/33247422/219342849-dceb2406-2b15-4abf-a760-dd8848493322.png)

**Reply by Aadesh-Baral** [reply]
This has been addressed in #5605.

**Comment by ramyaragupathy** [comment]
One more feedback on notification from the meeting today: let users receive notifications on task completion.
